### PR TITLE
Further improvements on the application form

### PR DESCRIPTION
**Pull request by viktorcsimma**
- There is now no default resident/extern status. However, both the finalization pages and the radio button itself requires a selection.
- Reworked the disclaimer for the Informatics workshop; the most notable change is that on the finalization page, it appears only for users who have indeed applied to the workshop.
- Language exams can be deleted.
- Other small changes.

### CODE REVIEW

**Comment by BertalanD** [comment]
hmm, the database seeder now seems to generate a null `applied_for_resident_status` for all aplicants.

File: resources/views/user/alfonso-language-exams.blade.php [hunk 1]
```diff
@@ -1,22 +1,44 @@
...
+                    <tr>
+                        <td><a target="_blank" href="/{{ $exam->path }}">{{ __('role.'.$exam->language) }}</a></td>
+                        <td>{{ $exam->level }}</td>
+                        <td>{{ $exam->type }}</td>
+                        <td>
+                            @if($exam->wasBeforeEnrollment and !isset($application))
+                            {{$exam->date->format('Y-m')}} (collegista státusz előtt szerezve)
+                            @else
+                            {{$exam->date->format('Y-m')}}
+                            @endif
```
Comment (BertalanD) [inline resources/views/user/alfonso-language-exams.blade.php]: nit: now that this is a full-width table, we could maybe also include the day (the day being chopped off kind of seems like a bug)

**Comment by viktorcsimma** [comment]
Oh; I've missed that... I'll fix it. Thanks!

File: app/Http/Controllers/Secretariat/UserController.php [hunk 2]
```diff
@@ -296,6 +297,28 @@ public function uploadLanguageExam(Request $request, User $user)
...
+     * @param User $user
+     * @param LanguageExam $exam
+     * @return RedirectResponse
+     */
+    public function deleteLanguageExam(Request $request, User $user, LanguageExam $exam)
+    {
+        $this->authorize('view', $user);
+        if ($exam->educationalInformation->user->id != $user->id) {
+            abort(400, 'The language exam does not belong to the given user.');
+        }
```
Comment (BertalanD) [inline app/Http/Controllers/Secretariat/UserController.php]: Not a PHP expert, feel free to disregard. But comparing database keys seems a bit fragile. Maybe something like

```suggestion
        if ($exam->educationalInformation->user->isNot($user)) {
            abort(400, 'The language exam does not belong to the given user.');
        }
```

**Comment by viktorcsimma** [comment]
Done;)

File: resources/views/auth/application/questions.blade.php [hunk 3]
```diff
@@ -48,15 +48,16 @@
...
-                                <input type="radio" name="status" value="resident"
+                                <input type="radio" name="status" value="resident" required
                                     {{ $checked ? 'checked' : '' }}>
                                 <span>@lang('role.resident')</span>
                             </label>
                         </p>
                         <p>
-                            @php $checked = old('status') ?  old('status') == 'extern' : !$user->application->applied_for_resident_status @endphp
+                            @php $checked = old('status') ?  old('status') == 'extern'
+                                    : (!is_null($user->application->applied_for_resident_status) && !$user->application->applied_for_resident_status) @endphp
```
Comment (BertalanD) [inline resources/views/auth/application/questions.blade.php]: I _understand_ what this code is doing, but I refuse to believe that handling a three-state flag is this convoluted in PHP.

**Review by BertalanD — COMMENTED** [review]
looks good overall, a couple things seemed odd to my (untrained) eye; feel free to ignore if they're not relevant

File: database/seeders/UsersTableSeeder.php [hunk 4]
```diff
@@ -132,7 +132,20 @@ private function createCollegist($user)
...
+        $submitted = rand(0, 1);
+        // submitted applications must not have a null status
+        if ($submitted) {
+            $appliedForResidentStatus = rand(0, 1);
+        } else {
+            $appliedForResidentStatus = rand(-1, 1);
+            if (-1 == $appliedForResidentStatus) {
+                $appliedForResidentStatus = null;
+            }
+        }
```
[the remark this answers is not in the document]
Comment (viktorcsimma) [reply database/seeders/UsersTableSeeder.php]: I'll add some more comments. Thanks!

File: resources/views/auth/application/questions.blade.php (hunk 3 above)
Comment (viktorcsimma) [reply resources/views/auth/application/questions.blade.php]: I've looked it up now; it seems `===` does the job.

File: app/Http/Controllers/Secretariat/UserController.php (hunk 2 above)
Comment (viktorcsimma) [reply app/Http/Controllers/Secretariat/UserController.php]: Nice catch:) I've not heard about this before.

File: resources/views/user/alfonso-language-exams.blade.php (hunk 1 above)
Comment (viktorcsimma) [reply resources/views/user/alfonso-language-exams.blade.php]: Right; done:)

**Comment by viktorcsimma** [comment]
Nice suggestions; thanks:) I've applied all of them.

**Review by BertalanD — APPROVED** [review]

**Pull request merged**